grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -283,6 +286,87 @@ class StockMovementController {
         render(template: "documents", model: [stockMovement: stockMovement])
     }
 
+    def comments = {
+        Requisition requisition = Requisition.get(params.id)
+        render(template: "comments", model: [requisition: requisition])
+    }
+
+    def addComment = {
+        Requisition requisition = Requisition.get(params.id)
+        StockMovement stockMovement = StockMovement.createFromRequisition(requisition)
+        [stockMovement: stockMovement, comment: new Comment()]
+    }
+
+    def editComment = {
+        Requisition requisition = Requisition.get(params.requisition)
+        StockMovement stockMovement = StockMovement.createFromRequisition(requisition)
+
+        Comment comment = Comment.get(params?.id)
+        if (!comment) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'comment.label', default: 'Comment'), comment.id])}"
+            redirect(action: "show", id: stockMovement?.id)
+            return
+        }
+        if (comment.sender.id != session.user.id) {
+            flash.message = "${g.message(code: 'auth.notAuthorized.message')}"
+            redirect(action: "show", id: stockMovement?.id)
+            return
+        }
+        render(view: "addComment", model: [stockMovement: stockMovement, comment: comment])
+    }
+
+    def deleteComment = {
+        Requisition requisition = Requisition.get(params.requisition)
+
+        Comment comment = Comment.get(params?.id)
+        if (!comment) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'comment.label', default: 'Comment'), params.id])}"
+            redirect(action: "show", id: requisition.id)
+            return
+        }
+
+        if (comment.sender.id != session.user.id) {
+            throw new UnsupportedOperationException("${g.message(code: 'errors.noPermissions.label')}")
+        }
+        requisitionService.deleteComment(comment, requisition)
+        flash.message = "${g.message(code: 'default.deleted.message', args: [g.message(code: 'comment.label', default: 'Comment'), comment.id])}"
+        redirect(action: "show", id: requisition.id)
+    }
+
+    def saveComment = {
+        Requisition requisition = Requisition.get(params.requisition)
+        StockMovement stockMovement = StockMovement.createFromRequisition(requisition)
+
+        Comment comment = new Comment(params)
+         if (comment.validate()) {
+             requisitionService.saveComment(comment)
```

Comment:
I need this save to get the comment id

Reply:
this save is needed anyway, because you create a **brand new** comment which is not yet managed, so nothing wrong with that :)